src/api/PDFDocument.ts

```diff
@@ -940,6 +944,47 @@ export default class PDFDocument {
     this.embeddedFiles.push(embeddedFile);
   }
 
+  private getRawAttachments() {
+    if (!this.catalog.has(PDFName.of('Names'))) return [];
+    const Names = this.catalog.lookup(PDFName.of('Names'), PDFDict);
+
+    if (!Names.has(PDFName.of('EmbeddedFiles'))) return [];
+    const EmbeddedFiles = Names.lookup(PDFName.of('EmbeddedFiles'), PDFDict);
+
+    if (!EmbeddedFiles.has(PDFName.of('Names'))) return [];
+    const EFNames = EmbeddedFiles.lookup(PDFName.of('Names'), PDFArray);
+
+    const rawAttachments = [];
+    for (let idx = 0, len = EFNames.size(); idx < len; idx += 2) {
+      const fileName = EFNames.lookup(idx) as PDFHexString | PDFString;
+      const fileSpec = EFNames.lookup(idx + 1, PDFDict);
+      rawAttachments.push({ fileName, fileSpec });
+    }
+
+    return rawAttachments;
+  }
+
+  /**
+   * Get all attachments that are embedded in this document.
+   *
+   * > **NOTE:** If you build a document with this library, this won't return
+   * > anything until you call [[save]] on the document.
+   *
+   * @returns Array of attachments with name and data
+   */
+  getAttachments() {
+    const rawAttachments = this.getRawAttachments();
+    return rawAttachments.map(({ fileName, fileSpec }) => {
+      const stream = fileSpec
+        .lookup(PDFName.of('EF'), PDFDict)
```

Comment:
Shouldn't we make sure "EF" and "F" key exist?

Reply:
That's a good question. These `lookup` and `of` functions don't ever return `undefined` in their signatures, and I'm not sure what the default value is if they don't find anything.